### ISSUE TITLE
Feature Hide Lower tier itens/unit

### ISSUE THREAD

**Issue by camarao69** [OpenThrone/OpenThrone#251]
I know it can just be myself but sometimes I buy the wrong  kind of item, can there be an option to hide lower tier itens in weapons, armor and units ?

**Comment by uaktags** [comment]
You thinking like a collapse/expand icon where you can hide the row/collapse it down?

I'd hate to hide anything, at least if you have >0 of those items still that haven't been converted/upgraded. But perhaps some sort of middle ground of collapsing from the normal height to a smaller height would be optimal?

![image](https://github.com/user-attachments/assets/5733f9f6-7725-41e5-bda3-8f043501b8ca)

**Comment by camarao69** [comment]
> You thinking like a collapse/expand icon where you can hide the row/collapse it down?
> 
> I'd hate to hide anything, at least if you have >0 of those items still that haven't been converted/upgraded. But perhaps some sort of middle ground of collapsing from the normal height to a smaller height would be optimal?
> 
> ![image](https://private-user-images.githubusercontent.com/893684/373729562-5733f9f6-7725-41e5-bda3-8f043501b8ca.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgwNjA0NjAsIm5iZiI6MTcyODA2MDE2MCwicGF0aCI6Ii84OTM2ODQvMzczNzI5NTYyLTU3MzNmOWY2LTc3MjUtNDFlNS1iZGEzLThmMDQzNTAxYjhjYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwNFQxNjQyNDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZDU5ZDdjNmIyODc3YWRmNmUzNmFiNjUzMjA1MmY5MWIzYWE3MDc5NjJiMmJiNTRiOTE4Y2VlNDllNzNhZjNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.xAvdnanF6mlck2KGjd8XPBTBvSJKfGyk0DlSDQTP8VE)

Exactly that, if it is more than 0 it should always appear 
But otherwise, and if it's the lowest tier, I thought it should be hidden, so as not to fill the page too much and to not create confusion.

**Comment by JOHNNYGAZER** [comment]
I like the idea of collapsing anything lower than the best you can access unless owned>0. The screenshot of the UI that you shared looks great as well @uaktags

**Comment by Revenate39** [comment]
I like your pic there Taco for armory items but not troops, I think that could stay expanded

**Comment by uaktags** [comment]
I agree, I think this really is a concern for Armory due to its size. Working on a new component for these tables to make this an option.

**Comment by uaktags** [comment]
![image](https://github.com/user-attachments/assets/d2277d90-4ea6-44bc-9344-2a0e87cba4d3)

this is the current logic:
- If you have Items, it'll be expanded
- If the item is the highest level item, it'll be expanded
- All others will be collapsed

**Comment by uaktags** [comment]
![image](https://github.com/user-attachments/assets/cde04567-43a5-442d-b969-05f0ec0cdb8c)
logic in action
